### ISSUE TITLE
Uninstall does not work for cent/rhel systems-regression

### ISSUE THREAD

**Issue by ShylajaDevadiga** [rancher/rke2#597]
Environment:
rke2 -v
rke2 version v1.18.13-rc1+rke2r1

Centos7.8
RHEL8.2

Describe the issue:
rke2-uninstall.sh does not delete all files.

```
/etc/yum.repos.d/rancher-rke2.repo
/etc/sysctl.d/60-rke2-cis.conf
/etc/sysconfig/rke2-server
/etc/selinux/targeted/active/modules/400/rke2
/etc/rancher/rke2
/etc/rancher/rke2/rke2.yaml
/var/lib/yum/repos/x86_64/7/rancher-rke2-1.18-testing
/var/lib/yum/repos/x86_64/7/rancher-rke2-common-testing
/usr/bin/rke2
/usr/bin/rke2-killall.sh
/usr/bin/rke2-uninstall.sh
/usr/share/rke2
/usr/lib/systemd/system/rke2-server.service
```

**Comment by davidnuzik** [comment]
RKE2 will still get killed. The problem is that expected files are not cleaned up; thus the node cannot be re-used. We will known-issue this in release notes for 1.18.13

**Comment by davidnuzik** [comment]
@ShylajaDevadiga nothing has changed in rke2-uninstall.sh since october 6th. Are you certain this is a new regression? You ran the script as root?

**Comment by ShylajaDevadiga** [comment]
Attaching log 
[uninstall.log](https://github.com/rancher/rke2/files/5690077/uninstall.log)

**Comment by ShylajaDevadiga** [comment]
@davidnuzik Same issue is seen in rke2 version v1.18.12+rke2r2 as well.

**Comment by davidnuzik** [comment]
Ok, not a regression. This will be marked as a known-issue in release notes.

**Comment by davidnuzik** [comment]
Assigned at least initially to @briandowns for initial investigation.
Bumped milestone out of v1.18.13+rke2r1 (our release we are trying to get out today). Put in the v1.19.5+rke2r1 milestone, but it should get fixed as soon as possible post v1.18.13 release since our dev-complete date for v1.19.5 is today. If this needs another day past dev-complete day this should be fine since it's such short notice to get a PR in for this.
We should not merge into master unless this is trivial to do because we need to get the 1.18.13 release out asap...

**Comment by davidnuzik** [comment]
The PR exists and is in review. I applied the Peer Review label.

**Comment by ShylajaDevadiga** [comment]
Uninstall hangs looking for /etc/sysconfig/rke2-server thats already removed. 

```
+ rm -f /etc/sysconfig/rke2-server
+ yum remove -y 'rke2-*'
Loaded plugins: fastestmirror
Resolving Dependencies
--> Running transaction check
---> Package rke2-common.x86_64 0:1.18.13~rc1~rke2r1-0.el7 will be erased
---> Package rke2-selinux.noarch 0:0.4-1.el7 will be erased
---> Package rke2-server.x86_64 0:1.18.13~rc1~rke2r1-0.el7 will be erased
--> Finished Dependency Resolution

Dependencies Resolved

===================================================================================================================
 Package              Arch           Version                            Repository                            Size
===================================================================================================================
Removing:
 rke2-common          x86_64         1.18.13~rc1~rke2r1-0.el7           @rancher-rke2-1.18-testing           153 M
 rke2-selinux         noarch         0.4-1.el7                          @rancher-rke2-common-testing          98 k
 rke2-server          x86_64         1.18.13~rc1~rke2r1-0.el7           @rancher-rke2-1.18-testing           679  

Transaction Summary
===================================================================================================================
Remove  3 Packages

Installed size: 153 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Erasing    : rke2-server-1.18.13~rc1~rke2r1-0.el7.x86_64                                                     1/3 
warning: file /etc/sysconfig/rke2-server: remove failed: No such file or directory
 
```

**Comment by Oats87** [comment]
Maybe we shouldn't be `rm -f /etc/sysconfig/rke2-server` because that is managed by the `rke2-server` RPM

**Comment by davidnuzik** [comment]
@briandowns could you review? Reopening

**Comment by briandowns** [comment]
PR inbound in a moment

**Comment by Oats87** [comment]
oups, we are calling the `rke2-uninstall.sh` using the `%preun` hook, leading to a loop of fun

**Comment by ShylajaDevadiga** [comment]
Validated using rke2 version v1.18.13-rc2+rke2r1 using (channel=testing) on Centos7.8, Centos8.2, Ubuntu20.04

- Installed version v1.18.13-rc2+rke2r1 
- Uninstalled using rke2-uninstall.sh
- Re-installed channel=stable to validate node is reusable.

[uninstall_rc2.log](https://github.com/rancher/rke2/files/5693353/uninstall_rc2.log)
